### DISCUSSION TITLE
Can ZTnet controller contact directly subnet that is running on?

### DISCUSSION THREAD

**Discussion by Urajmal** [sinamics/ztnet#201] [Q&A] [answered]
Hi! 

I am just asking if the controller can somehow passthrough the /24 of the subnet on which it is running on, for example: my controller has IP: 192.168.0.63 and I would like that all joined nodes can access on this network also 192.168.0.200 and 192.168.0.210? 

Or is it only standalone controller, and I have to add Mikrotik like router from the same network, and then input Route to 192.168.0.0/24?

Thank you for your answers!

**Comment by sinamics** [comment]
Hi
I guess you are using the standalone version without docker?
If so, just connect controller to a network using `zerotier-cli join ###`
It will then show up as the controller in ztnet status field. 

Now you can just add ip forward and make the neccessary routing in ztnet and all clients will have access to your local 192.168.0.x network as described here:
https://ztnet.network/usage/override_default_route

**Reply by Urajmal** [reply]
Thank you for fast answer! 

That is correct, I am runing Rpi Server on Debian Server. Controller internal IP is 192.168.0.63
I managed to activate IP Forwarding, added Controller and added Route in the ZeroTier.

I am unable to connect/ping to internal network of 192.168.0.0/24, via my phone. 
I am able to ping between nodes, and also controller IP. 
I am able to ping from Debian CLI to controller IP and to nodes IP.
I have attached a few pictures, for help.
![ZTconfig](https://github.com/sinamics/ztnet/assets/81016616/f8e8411b-a45d-470b-9ae5-19a105b09431)
![RoutesRpi](https://github.com/sinamics/ztnet/assets/81016616/37dad08b-87a4-4330-bd83-b94e1c16d852)

Did I miss any points from Override Default Route site?

**Reply by Urajmal** [reply]
One additional info. 

I am able to ping from nodes to Linux IP of controller: 192.168.0.63, but none of the other 0.x IPs.

**Reply by sinamics** [reply, marked as answer]
Seems correct to me. 
I just tested the exact same thing myself, and it works. I can ping all nodes on my local network from my cell phone.
Are you sure you got the iptables command correct?

verify that the `ZT_IFACE` is pointing to the correct NIC for your zerotier network.
https://ztnet.network/usage/override_default_route#step-2-configure-iptables-for-nat-and-forwarding--run-on-debian-client-

**Reply by Urajmal** [reply]
Thank you! 

Yes, I manage to get it to work, when I put in the commands for the iptables. 

Thank you again!

**Reply by Urajmal** [reply]
Ok thank you! 

Now my question regarding Docker version and not Debian stand alone version is, how can I add (enter in docker and use zerotier-cli command), Controller to the network?

And the second question is, how can I set up the iptables correctly when running Docker? In regarding to that I want to have accessible local network.

What is actually needed in host Linux machine, and what is needed in Docker itself?

Thank you again!

**Reply by sinamics** [reply]
> Now my question regarding Docker version and not Debian stand alone version is, how can I add (enter in docker and use zerotier-cli command), Controller to the network?

If you needs access to zerotier cli inside docker, you can login to the container and use the cli command. 
```bash
docker exec -it zerotier bash
```

> And the second question is, how can I set up the iptables correctly when running Docker? In regarding to that I want to have accessible local network.

Docker uses the `DOCKER-USER` chain. Example:
 ```bash
iptables -I DOCKER-USER -p tcp --dport 3000 -j ACCEPT
iptables -A DOCKER-USER -j DROP
```

Additionally, any ports that need to be exposed outside the Docker container must be explicitly specified in the `docker-compose.yml` file. Example:
```yml
  ztnet:
    image: sinamics/ztnet:latest
    container_name: ztnet
    ports:
      - 3000:3000    ( outside container : inside container )
```
